common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -303,6 +303,101 @@ def max_score(self):
         )
         return lcp.get_max_score()
 
+    def generate_report_data(self, user_state_iterator, limit_responses=None):
+        """
+        Return a list of student responses to this block in a readable way.
+
+        Arguments:
+            user_state_iterator: iterator over UserStateClient objects.
+                E.g. the result of user_state_client.iter_all_for_block(block_key)
+
+            limit_responses (int/None): maximum number of responses to include.
```

Comment:
nit: You can use the new convention of ``Optional[int]``

Reply:
What convention? I didn't find it in openedx; the documentation syntax seems rather free or depends on the section.
But I changed it to `int|None`, this may be better and it's used.

Reply:
I meant the new [Python type hints standard|https://www.python.org/dev/peps/pep-0484/#union-types], it's likely to be understandable by more tools in the future.

Reply:
But it's not to be used inside the documentation; for documentation there are other syntaxes but we don't seem to follow any. Also, it's Python 3.5.
We could just say „(int)“, there's already an explanation about None afterwards.

Reply:
In python2 it can be used inside documentation. This isn't exactly the format, but still, it's better to settle on the official standard I feel. Python2 is almost EOL soon so the code will be moving slowly to Python3 anyway. 
It's up to you though. I don't really see a strong convention in the Open edX codebase for this.

Reply:
In Open edX it seems it's left to personal opinion, and I don't have any on this. I'd leave `int|None` because I have seen `|` in other documentations in openedx